.github/workflows/reports.yml

```diff
@@ -490,3 +522,194 @@ jobs:
+      fail-fast: false
+      matrix:
+        include:
+          - project: { repo: AztecProtocol/aztec-packages, path: noir-projects/noir-protocol-circuits/crates/private-kernel-inner }
+          - project: { repo: AztecProtocol/aztec-packages, path: noir-projects/noir-protocol-circuits/crates/private-kernel-tail }
+          - project: { repo: AztecProtocol/aztec-packages, path: noir-projects/noir-protocol-circuits/crates/private-kernel-reset }
+          - project: { repo: AztecProtocol/aztec-packages, path: noir-projects/noir-protocol-circuits/crates/rollup-base-private }
+          - project: { repo: AztecProtocol/aztec-packages, path: noir-projects/noir-protocol-circuits/crates/rollup-base-public }
+
+    name: External repo execution report - ${{ matrix.project.repo }}/${{ matrix.project.path }}
+    steps:
+      - name: Download nargo binary
+        uses: actions/download-artifact@v4
+        with:
+          name: nargo
+          path: ./nargo
+
+      - name: Set nargo on PATH
+        run: |
+          nargo_binary="${{ github.workspace }}/nargo/nargo"
+          chmod +x $nargo_binary
+          echo "$(dirname $nargo_binary)" >> $GITHUB_PATH
+          export PATH="$PATH:$(dirname $nargo_binary)"
+          nargo -V
+
+      - uses: actions/checkout@v4
+        with:
+          path: scripts
+          sparse-checkout: |
+            test_programs/execution_report.sh
+          sparse-checkout-cone-mode: false
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ matrix.project.repo }}
+          path: test-repo
+          ref: ${{ matrix.project.ref }}
+
+      - name: Get package name
+        run: |
+          PACKAGE_NAME=$(basename ${{ matrix.project.path }})
+          echo $PACKAGE_NAME
+
+      - name: Download program artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: compilation_artifact_*
+          path: ./artifacts
+
+      - name: Move compilation artifacts
+        run: |
+          NAME_FROM_MATRIX=${{ matrix.project.path }}
+          NAME_FROM_MATRIX=$(basename $NAME_FROM_MATRIX)
+          # Convert `-` to `_` as that is how package names must be specified
+          NAME_FROM_MATRIX="${NAME_FROM_MATRIX//-/_}"
+          for artifact in ./artifacts/*; do
+            ARTIFACT_NAME=$(basename $artifact .json)
+            PACKAGE_NAME="${ARTIFACT_NAME#compilation_artifact_}"
+            if [[ "$PACKAGE_NAME" == "$NAME_FROM_MATRIX" ]]; then
+              mkdir ./test-repo/${{ matrix.project.path }}/target
+              mv ./artifacts/$ARTIFACT_NAME/$ARTIFACT_NAME.json ./test-repo/${{ matrix.project.path }}/target/$PACKAGE_NAME.json
+              break
+            fi
+          done
+
+      - name: Generate execution report
+        working-directory: ./test-repo/${{ matrix.project.path }}
+        run: |
+          mv /home/runner/work/noir/noir/scripts/test_programs/execution_report.sh ./execution_report.sh
```

Comment:
Is it worth just inlining this into the compilation report job so that it emits multiple report artifacts? That's what I'm doing in https://github.com/noir-lang/noir/pull/6832 plus it will make life easier once we tackle #6844

Reply:
Yeah it may be worth just inlining this, especially as the execution reports depends on the compilation report. We do have certain packages we compile but do not execute, such as `noir-contracts` as it is a library. I could add an extra flag to the matrix as to whether we should execute?

Reply:
Hmm, yeah the contracts are a little bit of an issue. I think adding a flag is the best option for now.

Reply:
I have inlined the execution report into the compilation report.